float32[6] pos